### PR TITLE
DOCS-1312: Fix Redirects

### PR DESCRIPTION
**Pull request by girishshankaran**
Fixes https://tigera.atlassian.net/browse/DOCS-1312

### CODE REVIEW

**Comment by girishshankaran** [comment]
@doucol Can you review this one?

**Review by doucol — CHANGES_REQUESTED** [review]
Same as other PR - you will likely need a force redirect here (301!)

Also, do you think you could simplify all of these with a smaller set of directives? Such as:
/archive/v3.22/* https://docs.tigera.io/archive/v3.22/:splat 301!
/archive/v3.23/* https://docs.tigera.io/archive/v3.23/:splat 301!

**Comment by girishshankaran** [comment]
> https://deploy-preview-431--calico-tigera.netlify.app

@doucol We already have these directives in the list:
/archive/v3.23/* https://docs.tigera.io/archive/v3.23/:splat 301
/archive/v3.22/* https://docs.tigera.io/archive/v3.22/:splat 301
/archive/v3.21/* https://docs.tigera.io/archive/v3.21/:splat 301
/archive/v3.20/* https://docs.tigera.io/archive/v3.20/:splat 301

The above directives seem to cover the entire list that I added. Therefore, I removed my list and added "!"to the following list:
/archive/v3.23/* https://docs.tigera.io/archive/v3.23/:splat 301!
/archive/v3.22/* https://docs.tigera.io/archive/v3.22/:splat 301!
/archive/v3.21/* https://docs.tigera.io/archive/v3.21/:splat 301!
/archive/v3.20/* https://docs.tigera.io/archive/v3.20/:splat 301!

If the source links are working, wont the redirection happen unless we do it forcefully?

File: _redirects [hunk 1]
```diff
@@ -63,6 +63,28 @@
...
+/archive/v3.24/getting-started/kubernetes/installation/config-options         https://docs.tigera.io/calico/3.24/getting-started/kubernetes/self-managed-onprem/config-options    301!
+/archive/v3.23/security/tutorials/kubernetes-policy-advanced                  https://docs.tigera.io/archive/v3.23/security/tutorials/kubernetes-policy-advanced                  301!
+/archive/v3.23/security/tutorials/app-layer-policy/enforce-policy-istio       https://docs.tigera.io/archive/v3.23/security/tutorials/app-layer-policy/enforce-policy-istio       301!
+/archive/v3.23/security/tutorials/kubernetes-policy-demo/kubernetes-demo      https://docs.tigera.io/archive/v3.23/security/tutorials/kubernetes-policy-demo/kubernetes-demo      301!
+/archive/v3.24/security/tutorials/protect-hosts                               https://docs.tigera.io/archive/v3.24/security/protect-hosts                                         301!
+/archive/v3.24/security/tutorials/kubernetes-policy-demo/kubernetes-demo      https://docs.tigera.io/archive/v3.24/security/tutorials/kubernetes-policy-demo/kubernetes-demo      391!
```
Comment (ctauchen) [inline _redirects]: s/391/301

**Review by ctauchen — APPROVED** [review]
One error, otherwise LGTM. Please merge after tests confirm these are cleared.

**Comment by ctauchen** [comment]
> Same as other PR - you will likely need a force redirect here (301!)
> 
> Also, do you think you could simplify all of these with a smaller set of directives? Such as: /archive/v3.22/* https://docs.tigera.io/archive/v3.22/:splat 301! /archive/v3.23/* https://docs.tigera.io/archive/v3.23/:splat 301!

We've got a cascade of rules, now in three layers:

1) The new page-specific redirects from SEO list
2) The directory-specific 200 proxy rules to maintain old URL manifest links (which were generally close enough, and which we're now refining)
3) The blanket 301 redirects to send all other old-docs traffic to the proxy locations under docs.tigera.io.

**Pull request merged**